static/src/js/containers/SavedProjectsContainer/__tests__/SavedProjectsContainer.test.js

```diff
@@ -1,103 +1,259 @@
+    SavedProjects.mockClear()
+    nock.cleanAll()
+    nock.disableNetConnect()
+  })
+
+  describe('When the component mounts with a valid authToken', () => {
+    test('fetches the projects and renders SavedProjects with projects', async () => {
+      const mockProjects = [
+        {
+          id: '123',
+          name: 'Project 1',
+          path: '/search?p=!C123',
+          created_at: '2022-01-01T00:00:00.000Z'
+        }
+      ]
+
+      nock(/localhost/)
+        .get(/projects/)
+        .reply(200, mockProjects)
+
+      setup()
+
+      await waitFor(() => {
+        const callWithLoadedData = SavedProjects.mock.calls.find(
+          (call) => call[0].isLoaded === true
+                     && call[0].isLoading === false
+                     && call[0].projects.length > 0
+        )
+        expect(callWithLoadedData).toBeTruthy()
+
+        if (callWithLoadedData) {
+          const [propsInCall] = callWithLoadedData
+          expect(propsInCall.isLoaded).toBe(true)
+          expect(propsInCall.isLoading).toBe(false)
+          expect(propsInCall.projects).toEqual(mockProjects)
+          expect(typeof propsInCall.onDeleteProject).toBe('function')
+        }
+      })
+    })
+  })
+
+  describe('When the authToken is null', () => {
+    test('renders SavedProjects with no projects', async () => {
+      setup({
+        overrideProps: {
+          authToken: null
+        }
       })
 
-      expect(enzymeWrapper.find(SavedProjects).length).toBe(1)
+      await waitFor(() => {
+        expect(SavedProjects).toHaveBeenCalled()
+        const lastCall = SavedProjects.mock.calls[SavedProjects.mock.calls.length - 1]
+        expect(lastCall[0].isLoading).toBe(false)
+        expect(lastCall[0].isLoaded).toBe(false)
+        expect(lastCall[0].projects).toEqual([])
+      })
+    })
+  })
+
+  describe('When deleting a project', () => {
+    const projectIdToDelete = '123'
+    const mockProjects = [
+      {
+        id: projectIdToDelete,
+        name: 'Project 1',
+        path: '/search?p=!C123',
+        created_at: '2022-01-01T00:00:00.000Z'
+      }
+    ]
+
+    beforeEach(() => {
+      nock(/localhost/)
+        .get(/projects/)
+        .reply(200, mockProjects)
+
+      window.confirm = jest.fn(() => true)
+    })
+
+    test('successfully deletes the project and shows a success toast', async () => {
+      nock(/localhost/)
+        .delete(`/projects/${projectIdToDelete}`)
+        .reply(204)
+
+      setup()
+
+      let onDeleteProjectCallback
+      await waitFor(() => {
+        const callWithLoadedData = SavedProjects.mock.calls.find(
+          (call) => call[0].isLoaded === true && call[0].projects.length > 0
+        )
+        expect(callWithLoadedData).toBeTruthy()
+        onDeleteProjectCallback = callWithLoadedData[0].onDeleteProject
+        expect(typeof onDeleteProjectCallback).toBe('function')
+      })
+
+      await act(async () => {
+        await onDeleteProjectCallback(projectIdToDelete)
```

Comment:
So for this we should try to delete as a user in here so finding the DOM element in this case its the (X) button and calling .click on that which in turn should call that function.

Since that is all being mocked I'm not sure I see the utility of that test in this container that kind of test should definitely be in the component

Reply:
SavedProjects shouldn't be mocked, PortalLinkContainer should be. Then you can click the button as a user correctly